ballista/scheduler/src/state/mod.rs

```diff
@@ -256,11 +260,46 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerState<T,
         plan: &LogicalPlan,
     ) -> Result<()> {
         let start = Instant::now();
+
+        // optimizing the plan here is redundant because the physical planner will do this again
+        // but it is helpful to see what the optimized plan will be
```

Comment:
Nice catch. How about changing this to the following:

`
        if log::max_level() >= log::Level::Debug {
            let optimized_plan = session_ctx.optimize(plan)?;
            debug!("Optimized plan: {}", optimized_plan.display_indent());
        }
`
so that we can avoid unnecessary optimization when the log level in set to be info.

Reply:
That's a good point. Maybe we just need to check the first path.